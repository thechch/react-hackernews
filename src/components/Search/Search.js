import PropTypes from 'prop-types';
import React, { useState } from 'react';

export const Search = ({
  initialValue = '',
  onSubmit,
  children = 'Search',
}) => {
  const [value, onChange] = useState(initialValue);

  const onFormSubmit = (event) => {
    event.preventDefault();
    onSubmit(value);
  };

  return (
    <form onSubmit={onFormSubmit}>
      {`${children} `}
      <input
        type="text"
        value={value}
        onChange={(event) => onChange(event.target.value)}
      />
      <button type="submit"> {children} </button>
    </form>
  );
};

Search.propTypes = {
  initialValue: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.string,
};
