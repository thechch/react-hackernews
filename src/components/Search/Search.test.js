import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search', () => {
  const props = {
    onChange: () => true,
    onSubmit: () => true,
    children: 'test',
  };
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search {...props}>Search</Search>, div);
  });

  test('snapshot', () => {
    const component = renderer.create(<Search {...props}>Search</Search>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays input', () => {
    const element = shallow(<Search {...props} />);
    expect(element.contains(<input />));
  });
});
