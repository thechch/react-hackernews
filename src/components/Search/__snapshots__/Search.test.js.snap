exports[`Search snapshot 1`] = `
<form
  onSubmit={[Function]}>
  Search 
   
  <input
    onChange={[Function]}
    type="text"
    value="" />
  <button
    type="submit">
    Search
  </button>
</form>
`;
